### PR TITLE
Merge release/2026.08 into develop (alpha6 follow-ups)

### PR DESCRIPTION
**Pull request by yingbull**
Forward-merge `release/2026.08` into `develop` after the `2026.08.0-alpha6`
release, keeping the branches aligned.

Brings to develop:
- **deb immutable-release fix** (#3529): attach the `.deb`s to the draft
  release, then publish — so future releases ship their Debian packages.
- **SRFax e2e tests** (#3528): `inbox-lifecycle`, `dedup-no-reimport`,
  `prescription-drugref`, plus the `STAGE_AS` staging wrapper and
  `csrfToken`/`postForm` helpers.

`pom` is kept at develop's `2026.09.0-SNAPSHOT` (`scm.tag` HEAD); only the
release-line's non-version content is merged.

Generated with Claude Code

## Summary by Sourcery

Finalize immutable releases with complete Debian packaging and broaden live fax and prescription integration coverage.

New Features:
- Add live SRFax and DrugRef end-to-end coverage for inbox lifecycle, inbound de-duplication, and prescription drug lookup workflows.

Bug Fixes:
- Ensure Debian packages are attached before immutable releases are published, preventing releases from becoming permanently incomplete.
- Prevent malformed short fax-line values from causing incorrect inbound fax de-duplication matches.

Enhancements:
- Improve fax data-access type safety and provide reusable authenticated form-posting and privileged file-staging support for live E2E tests.

Documentation:
- Document live fax E2E prerequisites, staging configuration, database access, and test execution order.

Tests:
- Add live tests for inbound fax lifecycle transitions, duplicate prevention, and DrugRef positive and negative lookups.
- Add regression coverage for malformed fax-line de-duplication values.

---
## Summary by cubic
Forward-merges `release/2026.08` into `develop`, fixing immutable-release packaging and expanding live SRFax/DrugRef coverage. Previously we published before attaching `.deb`s; now we publish only after `carlos-emr` and `carlos-emr-drugref` `.deb`s are attached, with idempotent skip when both are present and a fail-fast error if a release was published without `.deb`s.

- Draft-first flow: `publish-release` builds WAR/SBOM and creates a draft; `deb-packages` downloads the draft WAR, builds/attests/uploads both `.deb`s, then publishes and sets the latest pointer. Adds a global concurrency group to serialize finalization, and moves the latest-pointer decision here to avoid races.
- E2E (live, not CI): adds inbox lifecycle, inbound dedup no-reimport (validates `DEDUP_WAIT_MS`), and DrugRef search tests with `STAGE_AS`, `csrfToken`, and `postForm` helpers; tightens staging safety and distinguishes DrugRef service failure from valid empty results.
- Java: inbound-fax de-dup treats fax-line values with fewer than 10 digits as “no line,” preventing coincidental matches; adds a regression test. `FaxJobDaoImpl.findByFileName` now uses a `TypedQuery<FaxJob>`.
- `pom`: `develop` remains `2026.09.0-SNAPSHOT`; only non-version content merges.

**Rollout**
- To finalize a release: run `publish-release` to create the draft, then let `deb-packages` attach both `.deb`s and publish. If `deb-packages` fails, re-run `deb-packages` for the same tag to complete publishing.
- Already-published WAR-only releases cannot accept `.deb`s due to immutable releases; ship `.deb`s with the next release.

<sup>Written for commit 43a03276f027559bd7ce593f75439ecdf9308b8f. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/carlos-emr/carlos/pull/3530?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: .github/workflows/deb-packages.yml [hunk 1]
```diff
@@ -110,11 +119,38 @@ jobs:
...
+            # Fully finalized: published with its .debs. Nothing to do (and a
+            # re-upload would 422 against the immutable release).
+            echo "Release $RELEASE_TAG is already published with its .deb assets; nothing to do."
+            echo "SKIP=true" >> "$GITHUB_ENV"
+          else
+            # Published but missing .debs: an immutable release cannot accept
+            # new assets, so this can never be completed. Fail fast and loud
+            # rather than dying later on an opaque HTTP 422 from the upload.
+            echo "::error::Release $RELEASE_TAG is already published but has no .deb assets. Immutable releases cannot accept new assets, so the .debs cannot be attached. This release is permanently WAR-only; ship the .debs with the next release."
+            exit 1
```
[the remark this answers is not in the document]
Comment (yingbull) [reply .github/workflows/deb-packages.yml]: Fixed in #3531 (on release/2026.08): the skip guard now requires BOTH the carlos-emr and carlos-emr-drugref .deb assets (matched by package-name prefix), not just any two .debs.

File: .github/workflows/deb-packages.yml [hunk 2]
```diff
@@ -210,3 +251,39 @@ jobs:
...
+          # Every asset (WAR, SBOM, both .debs) is now attached to the draft.
+          # Flip it to published — this is the point the release becomes
+          # immutable. The latest-pointer decision lives here (not in
+          # publish-release.yml) because the release is only published once all
+          # assets, including these .debs, are in place.
+          prerelease=false
+          if [[ "$RELEASE_TAG" =~ -(alpha|beta|rc)[1-9][0-9]*$ ]]; then
+            prerelease=true
+          fi
+          release_line=$(sed -E 's/^([0-9]{4}\.[0-9]{2})\..*$/\1/' <<< "$RELEASE_TAG")
```
[the remark this answers is not in the document]
Comment (yingbull) [reply .github/workflows/deb-packages.yml]: Fixed in #3531 (on release/2026.08): removed the unused release_line assignment (SC2034).

File: scripts/e2e/fax/dedup-no-reimport.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+const DB = process.env.CARLOS_DB_NAME || 'oscar';
+function sql(q) {
+  const [cmd, ...pre] = MARIADB;
+  return execFileSync(cmd, [...pre, '-N', DB, '-e', q], { encoding: 'utf8' }).trim();
+}
+const must = (cond, msg) => { if (!cond) throw new Error(msg); };
+const digitsTail = (v) => (v || '').replace(/\D/g, '').slice(-10);
+
+// Two poll cycles at the scheduler's 60s cadence, plus margin.
+const WAIT_MS = parseInt(process.env.DEDUP_WAIT_MS || '150000', 10);
```
[the remark this answers is not in the document]
Comment (yingbull) [reply scripts/e2e/fax/dedup-no-reimport.js]: Fixed in #3531 (on release/2026.08): DEDUP_WAIT_MS is now validated as an integer >= 130000 (more than two 60s poll cycles); a NaN/short value throws instead of passing trivially.

File: scripts/e2e/fax/prescription-drugref.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  await p.waitForTimeout(600);
+  return p.evaluate((t) => {
+    const text = document.body.innerText || '';
+    // Result rows in ChooseDrug.jsp carry an "(Info)" affordance per drug.
+    const needle = t.toLowerCase();
+    const hits = [...document.querySelectorAll('a,tr,li')]
+      .map((e) => (e.textContent || '').trim())
+      .filter((s) => s.length < 120 && s.toLowerCase().includes(needle));
+    return { count: hits.length, sample: hits.slice(0, 3), failedOrEmpty: /unavailable|no results|not found|failed/i.test(text) || text.length < 400 };
+  }, term);
```
[the remark this answers is not in the document]
Comment (yingbull) [reply scripts/e2e/fax/prescription-drugref.js:55]: Fixed in #3531 (on release/2026.08): searchDrug now returns a separate `failed` flag (service error vs valid empty). The negative case must have COMPLETED without failure and returned zero, so a DrugRef error page can no longer satisfy it. Verified live: negative now reports "lookup completed, no results".

File: .github/workflows/deb-packages.yml [hunk 5]
```diff
@@ -210,3 +251,39 @@ jobs:
...
+
```
[the remark this answers is not in the document]
Comment (yingbull) [reply .github/workflows/deb-packages.yml]: Fixed in #3531 (on release/2026.08): removed the unused release_line assignment from the publish step.

File: .github/workflows/publish-release.yml [hunk 6]
```diff
@@ -393,6 +370,6 @@ jobs:
           if ! gh workflow run deb-packages.yml --repo "$REPO" \
                  --ref "refs/tags/$RELEASE_TAG" \
                  -f "tag=$RELEASE_TAG"; then
```
[the remark this answers is not in the document]
Comment (yingbull) [reply .github/workflows/publish-release.yml:372]: Acknowledged. The critical latest-pointer race is handled by deb-packages.yml now using a single GLOBAL concurrency group, which serializes the actual publish/latest decision across releases. The residual this raises — the release-notes start-tag for two SAME-line releases published within the ~10-min deb-build window — is cosmetic (affects only the generated notes range), and same-line releases that close together are not part of the deliberate release flow. Holding the publish-release lock across a separately-dispatched workflow is not achievable, so this is intentionally left as a documented minor limitation rather than over-engineered.
